src/rgw/rgw_sal_motr.cc

```diff
@@ -2115,52 +2115,77 @@ int MotrObject::get_bucket_dir_ent(const DoutPrefixProvider *dpp, rgw_bucket_dir
 
     rgw_bucket_dir_entry ent_to_check;
 
+    // Check entry in the cache
     if (this->store->get_obj_meta_cache()->get(dpp, this->get_name(), bl) == 0) {
-      iter = bl.cbegin();
-      ent_to_check.decode(iter);
-      if (ent_to_check.is_current()) {
+        iter = bl.cbegin();
+        ent_to_check.decode(iter);
         ent = ent_to_check;
         rc = 0;
         goto out;
-      }
-    }
-
-    ldpp_dout(dpp, 20) <<__func__<< ": versioned bucket!" << dendl;
-    keys[0] = this->get_name();
-    rc = store->next_query_by_name(bucket_index_iname, keys, vals);
-    if (rc < 0) {
-      ldpp_dout(dpp, 0) << __func__ << "ERROR: NEXT query failed. " << rc << dendl;
-      return rc;
     }
 
-    rc = -ENOENT;
-    for (const auto& bl: vals) {
-      if (bl.length() == 0)
-        break;
+    if(this->have_instance())
+    {
+      // TODO : Handle null version-id scenarios
 
-      iter = bl.cbegin();
-      ent_to_check.decode(iter);
-      if (ent_to_check.is_current()) {
-        ldpp_dout(dpp, 20) <<__func__<< ": found current version!" << dendl;
+      // Cache miss.
+      rc = this->store->do_idx_op_by_name(bucket_index_iname,
+                          M0_IC_GET, this->get_key().to_str(), bl);
+      if(rc<0)
+        return rc;
+      else
+      {
+        iter = bl.cbegin();
+        ent_to_check.decode(iter);
         ent = ent_to_check;
```

Comment:
now that ent is not getting assigned with ent_to_check on any condition, probably we can avoid to variables and decode iter in ent directly.

Reply:
yes, done. Decoded the iter in ent directly.